src/server/game/Guilds/Guild.cpp

```diff
@@ -1504,26 +1504,45 @@ void Guild::HandleSetEmblem(WorldSession* session, const EmblemInfo& emblemInfo)
     }
 }
 
-void Guild::HandleSetNewGuildMaster(WorldSession* session, std::string const& name)
+void Guild::HandleSetNewGuildMaster(WorldSession* session, std::string const& name, bool isSelfPromote)
 {
     Player* player = session->GetPlayer();
-    // Only the guild master can throne a new guild master
-    if (!_IsLeader(player))
-        SendCommandResult(session, GUILD_COMMAND_CHANGE_LEADER, ERR_GUILD_PERMISSIONS);
-    // Old GM must be a guild member
-    else if (Member* oldGuildMaster = GetMember(player->GetGUID()))
-    {
-        // Same for the new one
-        if (Member* newGuildMaster = GetMember(name))
-        {
-            _SetLeaderGUID(newGuildMaster);
+    Member* oldGuildMaster;
+    Member* newGuildMaster;
 
-            SQLTransaction trans(nullptr);
-            oldGuildMaster->ChangeRank(trans, GR_INITIATE);
+    if (isSelfPromote)
+    {
+        oldGuildMaster = GetMember(GetLeaderGUID());
+        newGuildMaster = GetMember(player->GetGUID());
+        if (!oldGuildMaster || !newGuildMaster)
+            return;
 
-            SendEventNewLeader(newGuildMaster, oldGuildMaster);
+        if (!newGuildMaster->IsRankNotLower(GR_MEMBER) || uint32(oldGuildMaster->GetInactiveDays()) < GUILD_MASTER_DETHRONE_DAYS)
```

Comment:
isnt this officers only?

Reply:
Not quite: https://us.battle.net/support/en/article/10333
"Eligible members must be guild rank 1, 2, or 3 and must have logged in within the last week."